src/apps/unagii/ethereum/unagii.vault.token-fetcher.ts

```diff
@@ -1,126 +1,62 @@
+  async getDefinitions({ multicall }): Promise<UnagiiVaultTokenDefinition[]> {
+    const vaultManagerAddresses = [
+      '0x7f75d72886d6a8677321e5602d18948abcb4281a',
+      '0xb088d7c71ea9ebaed981c103fc3019b59950d2c9',
+      '0x8ef11c51a666c53aeeec504f120cd1435e451342',
+    ];
+
+    const vaultDefinitions = await Promise.all(
+      vaultManagerAddresses.map(async vaultAddress => {
+        const vaultManagerContract = this.contractFactory.unagiiV2Vault({
+          address: vaultAddress,
+          network: this.network,
+        });
+        const [tokenAddressRaw, underlyingTokenAddressRaw] = await Promise.all([
+          multicall.wrap(vaultManagerContract).uToken(),
+          multicall.wrap(vaultManagerContract).token(),
         ]);
-        const underlyingTokenAddress =
-          underlyingTokenAddressRaw.toLowerCase() === ETH_ADDR_ALIAS
-            ? ZERO_ADDRESS
-            : underlyingTokenAddressRaw.toLowerCase();
-        const underlyingToken = await this.appToolkit.getBaseTokenPrice({ network, address: underlyingTokenAddress });
-        if (!underlyingToken) return null;
-
-        // Determine total supply
-        const underlyingPrice = underlyingToken.price || 0;
-        const supply = Number(totalSupplyRaw) / 10 ** decimals;
-        const underlyingAssets = Number(totalAssetsRaw) / 10 ** underlyingToken.decimals;
-
-        // Determine the price per share
-        const pricePerShare = underlyingAssets / supply;
-        const price = pricePerShare * underlyingToken.price;
-
-        const reserve = supply * pricePerShare;
-        const liquidity = reserve * underlyingPrice;
-
-        const tokens = [{ ...underlyingToken, reserve }];
-        const reservePercentages = tokens.map(t => reserve * (t.price / liquidity));
-        const ratio = reservePercentages.map(p => `${Math.round(p * 100)}%`).join(' / ');
-        const secondaryLabel = ratio;
-
-        const displayProps = {
-          label: symbol,
-          secondaryLabel,
-          images: [getTokenImg(underlyingToken.address)],
-          statsItems: [
-            {
-              label: 'Liquidity',
-              value: buildDollarDisplayItem(liquidity),
-            },
-            {
-              label: 'Supply',
-              value: buildNumberDisplayItem(supply),
-            },
-            {
-              label: 'Ratio',
-              value: ratio,
-            },
-          ],
-        };
-
-        const dataProps = {
-          liquidity,
-          isActive: vault.isActive,
-        };
-
-        const token: AppTokenPosition = {
-          address: tokenAddress,
-          type: ContractType.APP_TOKEN,
-          network,
-          appId,
-          groupId,
-          symbol,
-          decimals,
-          supply,
-          price,
-          pricePerShare,
-          tokens,
-          dataProps,
-          displayProps,
+        return {
+          address: vaultAddress,
+          tokenAddress: tokenAddressRaw.toLowerCase(),
```

Comment:
The `address` should be the vault token address. You can keep the vault address in the definition separately as `vaultAddress`. Please be sure to update the generic too, it should be the `UnagiiUToken` not the `UnagiiV2Vault` (think: the token should always be an ERC20; the vault is not the ERC20!)